FreshRSS/FreshRSS
# Improved: Fetch articles with selector but do not delete the class attribute. (Simplepie: new method: rename_attribute)

# My User Story:
As a reader I want to configure the content of fetched articles, so that I can hide parts that I do not need.

# Changes proposed in this pull request:
## User side:
- the `id` and `class` attributes will not deleted anymore. It will renamed to `data-sanitized-id` / `data-sanitized-class`
- with the extension CustomCSS, it is possible to style the `data-sanitized-class` tags (f.e. `div[data-feed="XX"] div[data-sanitized-class*="CLASSNAME"]`

Here an example, that the fetched article footer could be re-styled with CSS (here with `background-color:red`)
![grafik](https://user-images.githubusercontent.com/1645099/151250087-65694d1d-ebe8-43ea-9c88-6e80b35773c7.png)

## technical side:
- added the method `rename_attr()`
- default of `$rename_attributes` is an empty array
- for FreshRSS `$simplePie->rename_attributes('class');` is set, so that the `class` attribute is renamed to `data-sanitized-class`

How to test the feature manually:

1. fetch articles with a selector
2. see the HTML
3. see the `data-sanitized-class`
4. do some CSS magic to style it

Pull request checklist:

- [ ] if you like it, I would do a PR to the SimplePie repository
